api.planx.uk/modules/auth/strategy/google.ts

```diff
@@ -0,0 +1,23 @@
+import { Strategy as GoogleStrategy } from "passport-google-oauth20";
+import { buildJWT } from "../service";
+
+export const googleStrategy = new GoogleStrategy(
+  {
+    clientID: process.env.GOOGLE_CLIENT_ID!,
+    clientSecret: process.env.GOOGLE_CLIENT_SECRET!,
+    callbackURL: `${process.env.API_URL_EXT}/auth/google/callback`,
+  },
+  async function (_accessToken, _refreshToken, profile, done) {
+    const { email } = profile._json;
+    const jwt = await buildJWT(email);
+
+    if (!jwt) {
+      return done({
+        status: 404,
+        message: `User (${email}) not found.Do you need to log in to a different Google Account?`,
+      } as any);
+    }
+
+    done(null, { jwt });
+  },
+);
```

Comment:
Can you share a bit about why you chose to organise this into `strategy/google` rather than adding into controller or service.ts ? Do you imagine `strategy/` will apply to more modules besides auth going forward?

Reply:
This is a great question.

It was really moved here just for organisation - it's part of the `auth` module, but not really a middleware or controller, just a "strategy" used by `Passport`. I'm not 100% sure but I think we'll end up with another [Passport Strategy](https://www.passportjs.org/concepts/authentication/strategies/) here for Microsoft SSO.

It might be simpler / more meaningful to group all `passport` logic into `/auth/passport` or something in future as opposed to having some `app.use()` functions using `passport` in `server.ts`

Reply:
That's helpful context, thanks ! Let's leave as this for now but maybe revisit organisation in future as needed ?